### DISCUSSION TITLE
used macro doesn't get suggestion

### DISCUSSION THREAD

**Discussion by DailyShana** [James-Yu/LaTeX-Workshop#4311] [Q&A]
![2024-07-03_174346](https://github.com/James-Yu/LaTeX-Workshop/assets/8469304/5abbf249-e0fa-40c8-94cb-372c47968574)

as shown in the above screen shot, there is a `\aaabbb`, and then type `\aaa` the `\aaabbb` is not shown in the suggestion list.

before version 9.12.0, the `\aaabbb` will shown as expected:
![image](https://github.com/James-Yu/LaTeX-Workshop/assets/8469304/e6c1ad9c-c53e-45bf-a59b-7619af649cb9)

**Comment by James-Yu** [comment]
You need to define it with macros like `newcommand` before the extension parses it.

If the macros are provided by your document class, there is another config to point to a new command definition file.

**Comment by jlelong** [comment]
I remember that the extension used to parse latex files for macros appearing in the body of the document. This part is still present in https://github.com/James-Yu/LaTeX-Workshop/blob/707b0585c6df96acfc8a57b9ef620f083c7e098c/src/completion/completer/macro.ts#L309-L315 but has somehow disappeared from https://github.com/James-Yu/LaTeX-Workshop/blob/707b0585c6df96acfc8a57b9ef620f083c7e098c/src/completion/completer/macro.ts#L217

I think this is what @DailyShana refers to. Is it an intentional removal or should we work on implementing it back?

**Reply by James-Yu** [reply]
This is because that `unified` won’t absorb unknown arguments for arbitrary macro, unless predefined. It make create argument-less suggestions, may not be quite useful. In the fallback logic, the behavior was retained. I’m not sure if such partial support would be a good idea.

**Reply by jlelong** [reply]
Thanks for the explanation. I will update the wiki https://github.com/James-Yu/LaTeX-Workshop/wiki/Intellisense#commands-starting-with- (second item)

**Reply by DailyShana** [reply]
> This is because that `unified` won’t absorb unknown arguments for arbitrary macro, unless predefined. It make create argument-less suggestions, may not be quite useful. In the fallback logic, the behavior was retained. I’m not sure if such partial support would be a good idea.

I think suggestions without argument is better than no suggestions. we can explain the reason in the wiki
